src/datasource.ts

```diff
@@ -0,0 +1,109 @@
+import { DataQueryRequest, DataSourceApi, DataSourceInstanceSettings, toDataFrame } from '@grafana/data';
+import { map, merge, catchError, of, lastValueFrom } from 'rxjs';
+
+import { MyQuery, MyDataSourceOptions } from './types';
+import _ from 'lodash';
+import { getBackendSrv } from '@grafana/runtime';
+
+export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
+  url?: string;
+
+  constructor(instanceSettings: DataSourceInstanceSettings<MyDataSourceOptions>) {
+    super(instanceSettings);
+
+    this.url = instanceSettings.url;
+  }
+
+  fetchMetric(realm: string, metricParams: Record<string, any>) {
+    if (realm) {
+      const result = getBackendSrv().fetch({
+        method: 'GET',
+        url: `${this.url}/api/${realm}/api/m`,
+        params: {
+          // Remove points without enough hosts reporting
+          removeincomplete: 1,
+          omitpoints: 'null',
+          ...metricParams,
+        },
+      });
+
+      return result;
+    }
+    return of();
+  }
+
+  query(options: DataQueryRequest<MyQuery>) {
+    const { range } = options;
+    const from = range!.from.valueOf();
+    const to = range!.to.valueOf();
+    const rangeMs = options.range.to.valueOf() - options.range.from.valueOf();
+    const sample = Math.max(10000, options.maxDataPoints ? Math.round(rangeMs / options.maxDataPoints) : 0);
```

Comment:
shouldn't sample be targetPoints here?

Reply:
I just copied this logic from the old datasource code - do you think targetpoints makes more sense?

Reply:
changed to intervalMs